python/ray/tune/impl/tuner_internal.py

```diff
@@ -143,9 +135,24 @@ def __init__(
         # without allowing for checkpointing tuner and trainable.
         # Thus this has to happen before tune.run() so that we can have something
         # to restore from.
-        experiment_checkpoint_path = Path(self._local_experiment_dir, _TUNER_PKL)
-        with open(experiment_checkpoint_path, "wb") as fp:
-            pickle.dump(self.__getstate__(), fp)
+        self._run_config.name = (
+            self._run_config.name
+            or StorageContext.get_experiment_dir_name(self.converted_trainable)
+        )
+        storage = StorageContext(
+            storage_path=self._run_config.storage_path,
+            experiment_dir_name=self._run_config.name,
+            storage_filesystem=self._run_config.storage_filesystem,
+        )
+        # TODO(justinvyu): Temporary fix for `get_experiment_checkpoint_dir`
```

Comment:
I mostly just don't want to add another attribute to the tuner, and want to just construct this with run_config.storage_path + run_config.name after I set the `storage_path` immediately rather than at runtime in the next PR.

See https://github.com/ray-project/ray/pull/43369#discussion_r1503415049

Reply:
Can actually remove this logic.